Sources/NIO/Utilities.swift

```diff
@@ -41,7 +52,37 @@ public enum System {
     ///
     /// - returns: The logical core count on the system.
     public static var coreCount: Int {
+#if os(Windows)
+        var dwLength: DWORD = 0
+        _ = GetLogicalProcessorInformation(nil, &dwLength)
+
+        let alignment: Int =
+            MemoryLayout<SYSTEM_LOGICAL_PROCESSOR_INFORMATION>.alignment
+        let pBuffer: UnsafeMutableRawPointer =
+            UnsafeMutableRawPointer.allocate(byteCount: Int(dwLength),
+                                             alignment: alignment)
+        defer {
+            pBuffer.deallocate()
+        }
+
+        let dwSLPICount: Int =
+            Int(dwLength) / MemoryLayout<SYSTEM_LOGICAL_PROCESSOR_INFORMATION>.stride
+        let pSLPI: UnsafeMutablePointer<SYSTEM_LOGICAL_PROCESSOR_INFORMATION> =
+            pBuffer.bindMemory(to: SYSTEM_LOGICAL_PROCESSOR_INFORMATION.self,
+                               capacity: dwSLPICount)
+
+        let bResult: Bool = GetLogicalProcessorInformation(pSLPI, &dwLength)
+        assert(bResult, "GetLogicalProcessorInformation: \(GetLastError())")
```

Comment:
```suggestion
        precondition(bResult, "GetLogicalProcessorInformation: \(GetLastError())")
```

Accessing the memory if this does not return `true` leads to reading uninitialised memory, which is undefined behaviour. We must crash here.